core/src/main/java/hudson/security/TokenBasedRememberMeServices2.java

```diff
@@ -303,39 +307,7 @@ private String findRememberMeCookieValue(HttpServletRequest request, HttpServlet
             return null;
         }
     }
-
-    @Override
-    protected Cookie makeValidCookie(String tokenValueBase64, HttpServletRequest request, long maxAge) {
-        Cookie cookie = super.makeValidCookie(tokenValueBase64, request, maxAge);
-        secureCookie(cookie, request);
-        return cookie;
-    }
-
-    @Override 
-    protected Cookie makeCancelCookie(HttpServletRequest request) {
-        Cookie cookie = super.makeCancelCookie(request);
-        secureCookie(cookie, request);
-        return cookie;
-    }
-    
-    /**
-     * Force always the http-only flag and depending on the request, put also the secure flag.
-     */
-    private void secureCookie(Cookie cookie, HttpServletRequest request){
-        // if we can mark the cookie HTTP only, do so to protect this cookie even in case of XSS vulnerability.
-        if (SET_HTTP_ONLY!=null) {
-            try {
-                SET_HTTP_ONLY.invoke(cookie,true);
-            } catch (IllegalAccessException | InvocationTargetException e) {
-                // ignore
-            }
-        }
-
-        // if the user is running Jenkins over HTTPS, we also want to prevent the cookie from leaking in HTTP.
-        // whether the login is done over HTTPS or not would be a good enough approximation of whether Jenkins runs in
-        // HTTPS or not, so use that.
-        cookie.setSecure(request.isSecure());
```

Comment:
Seems to be obsolete: https://github.com/spring-projects/spring-security/blob/56928f61f0f0dbba8dc5c7c6a99746abc39f8230/web/src/main/java/org/springframework/security/web/authentication/rememberme/AbstractRememberMeServices.java#L422-L429

Reply:
IIRC setHttpOnly reflection was due to previous Tomcat version, with the current one supported by Jenkins, it's no longer necessary as we are using Servlet 3.0+.